paprika/setup.py

```diff
@@ -31,732 +26,685 @@ def _get_installed_benchmarks():
+    #         if generate_gaff_files:
+    #             generate_gaff(mol2_file=self.benchmark_path.joinpath(
+    #             self.host_yaml["structure"]),
+    #                           residue_name=self.host_yaml["resname"],
+    #                           output_name=self.host,
+    #                           directory_path=self.directory,
+    #                           gaff=gaff_version)
+    #             if guest:
+    #                 generate_gaff(mol2_file=self.benchmark_path.joinpath(
+    #                     self.guest).joinpath(self.guest_yaml["structure"]),
+    #                               output_name=self.guest,
+    #                               residue_name=self.guest_yaml["name"],
+    #                               directory_path=self.directory,
+    #                               gaff=gaff_version)
+    #     if not build:
+    #         self.populate_window_list(input_pdb=os.path.join(self.directory,
+    #         f"{self.host}-{self.guest}.pdb" if self.guest is not None
+    #         else f"{self.host}.pdb"))
+
+    # def parse_yaml(self, installed_benchmarks, guest_orientation):
+    #     """
+    #     Read the YAML recipe for the host and guest.
+    #
+    #     Returns
+    #     -------
+    #
+    #     """
+    #     try:
+    #         if guest_orientation:
+    #
+    #             host_yaml = installed_benchmarks["host_guest_systems"][self.host]
+    #             ["yaml"][guest_orientation]
+    #         else:
+    #             host_yaml = installed_benchmarks["host_guest_systems"][self.host]
+    #             ["yaml"]["p"]
+    #
+    #     except KeyError:
+    #         logger.error(f"Cannot find YAML recipe for host: {self.host}")
+    #         logger.debug(installed_benchmarks)
+    #         raise FileNotFoundError
+    #     try:
+    #         guest_yaml = installed_benchmarks["host_guest_systems"][self.host]
+    #         [self.guest]
+    #     except KeyError:
+    #         if self.guest == "release":
+    #             guest_yaml = None
+    #         else:
+    #             logger.error(f"Cannot find YAML recipe for guest: {self.guest}")
+    #             logger.debug(installed_benchmarks)
+    #             raise FileNotFoundError
+    #
+    #     return host_yaml, guest_yaml
+
+    @classmethod
+    def build_static_restraints(
+        cls,
+        coordinate_path: str,
+        n_attach_windows: Optional[int],
+        n_pull_windows: Optional[int],
+        n_release_windows: Optional[int],
+        restraint_schemas: List[Dict[str, Any]],
+        use_amber_indices: bool = False,
+    ) -> List[DAT_restraint]:
+        """
+
+        Parameters
+        ----------
+        coordinate_path
+        n_attach_windows
+        n_pull_windows
+        n_release_windows
+        restraint_schemas
+
+            This should be a dictionary of the form:
+
+                * atoms: List[str]
+                * force_constant: float
+
```

Comment:
IIUC, there is no way to tell which phase and window a given `restraint_schema` is associated with. Is this coming from Evaluator?

Reply:
It is yes.